src/mesh/nemesis_io_helper.C

```diff
@@ -2660,77 +2675,90 @@ Nemesis_IO_Helper::write_element_values(const MeshBase & mesh,
-                                 local_soln_buffer.begin(), [](Number n) { return n.real(); });
-                  ex_err = exII::ex_put_elem_var(ex_id,
-                                                 timestep,
-                                                 static_cast<int>(stride*v+1),
-                                                 static_cast<int>(sbd_id),
-                                                 static_cast<int>(local_soln.size()),
-                                                 local_soln_buffer.data());
-                  EX_CHECK_ERR(ex_err, "Error writing element real values.");
-
-                  std::transform(local_soln.begin(), local_soln.end(),
-                                 local_soln_buffer.begin(), [](Number n) { return n.imag(); });
-                  ex_err = exII::ex_put_elem_var(ex_id,
-                                                 timestep,
-                                                 static_cast<int>(stride*v+2),
-                                                 static_cast<int>(sbd_id),
-                                                 static_cast<int>(local_soln.size()),
-                                                 local_soln_buffer.data());
-                  EX_CHECK_ERR(ex_err, "Error writing element imaginary values.");
-
-                  if (write_complex_abs)
-                    {
-                      std::transform(local_soln.begin(), local_soln.end(),
-                                     local_soln_buffer.begin(), [](Number n) { return std::abs(n); });
-                      ex_err = exII::ex_put_elem_var(ex_id,
-                                                     timestep,
-                                                     static_cast<int>(stride*v+2),
-                                                     static_cast<int>(sbd_id),
-                                                     static_cast<int>(local_soln.size()),
-                                                     local_soln_buffer.data());
-                      EX_CHECK_ERR(ex_err, "Error writing element magnitudes.");
-                    }
+                    int stride = write_complex_abs ? 3 : 2;
+                    std::vector<Real> local_soln_buffer(local_soln.size());
+                    std::transform(local_soln.begin(), local_soln.end(),
+                                   local_soln_buffer.begin(), [](Number n) { return n.real(); });
+                    ex_err = exII::ex_put_elem_var(ex_id,
+                                                   timestep,
+                                                   static_cast<int>(stride*(var_ctr+comp)+1),
+                                                   static_cast<int>(sbd_id),
+                                                   static_cast<int>(local_soln.size()),
+                                                   local_soln_buffer.data());
+                    EX_CHECK_ERR(ex_err, "Error writing element real values.");
```

Comment:
IIRC Exodus can be picky about 0-length data in some APIs, but I can't remember whether the preferred solution is "make sure you write a 0-length vector" or "make sure you *don't* write a 0-length vector", nor can I remember which APIs were the fragile ones.

Could we get some test coverage of this case, perhaps by just assigning every element in your unit test to a different subdomain, so that as soon as we test with `mpiexec -np 2` we'll end up with subdomains that have 0 elements on one processor or the other?

Reply:
I don't think we would make it this far if there were zero elements on the subdomain.

Reply:
Yeah, looks like we'd be short-circuiting at the `elem_ids.size()` above.  But that just pushes the potential failure point elsewhere; I'd still feel a lot safer with a test case that hits it and doesn't fail.